cpp/src/io/parquet/page_data.cu

```diff
@@ -994,7 +994,10 @@ static __device__ bool setupLocalPageInfo(page_state_s* const s,
   int chunk_idx;
 
   // Fetch page info
-  if (!t) s->page = *p;
+  if (!t) {
+    s->page         = *p;
+    s->nesting_info = nullptr;
```

Comment:
So IIUC this is the key change that stops it from looking in the decode cache?

Reply:
Oops, I didn't update the description.
This null prevents `null_count_back_copier` from copying `nesting_decode_cache` to `nesting_decode` when the nesting info hasn't been set up because of an early exit from `setupLocalPageInfo`.